src/net35/Radical/Observers/BrokerObserver.cs

```diff
@@ -77,9 +80,9 @@ public MessageBrokerMonitor( IMessageBroker broker )
         /// </summary>
         /// <typeparam name="TMessage">The type of the message.</typeparam>
         /// <returns>This monitor instance.</returns>
-        public MessageBrokerMonitor WaitingFor<TMessage>() where TMessage : class
+        public MessageBrokerMonitor WaitingFor<TMessage>( InvocationModel invocationModel = InvocationModel.Default ) where TMessage : class
```

Comment:
Unfortunately you need to recompile in order to be able to call that method the methods virtual table is changed in the destination assembly since the signature is indeed changed, so the calling assembly cannot call the destination without a recompile. From the `SemVer` perspective it is a breaking change.

This specific scenario is very interesting, say that you just found the bug in production, you want to be able to release the new version of `Radical` without recompiling and releasing your entire application.

More details here: https://lostechies.com/jimmybogard/2010/05/18/caveats-of-c-4-0-optional-parameters/